[BUG] Documentation: TypeScript in Custom Extensions section fails to compile

### Software versions

Python version        :  3.12.2 (main, Feb 10 2024, 11:36:40) [Clang 14.0.3 (clang-1403.0.22.14.1)]
IPython version       :  (not installed)
Tornado version       :  6.4
Bokeh version         :  3.3.4
BokehJS static path   :  /Users/mdickinson/Desktop/.venv/lib/python3.12/site-packages/bokeh/server/static
node.js version       :  v20.11.1
npm version           :  10.2.4
jupyter_bokeh version :  (not installed)
Operating system      :  macOS-13.6.4-x86_64-i386-64bit

### Browser name and version

Not applicable

### Jupyter notebook / Jupyter Lab version

Not applicable

### Expected behavior

Steps to reproduce:

* Create a Python 3.12 venv, activate it, and do a `pip install bokeh` inside the activated environment.
* Starting from https://docs.bokeh.org/en/latest/docs/user_guide/advanced/extensions.html:
  *  copy the TypeScript code from https://docs.bokeh.org/en/latest/docs/user_guide/advanced/extensions.html#javascript-models-and-views to a file `custom.ts`
  * copy the first Python snippet from https://docs.bokeh.org/en/latest/docs/user_guide/advanced/extensions.html#putting-it-together to a file `custom.py`
  * copy the second Python snippet from the same section to a file `custom_extension.py`
  * place all three files (`custom.py`, `custom.ts`, `custom_extension.py`) within an otherwise empty directory
* Edit the `custom_extension.py` file to add an import line: `from custom import Custom`
* From within the directory containing the three files above, execute `bokeh serve custom_extension.py --show`

Expected behaviour: an example web app is brought up in the browser that looks similar to the screenshot in the documentation.


### Observed behavior

The server fails with a traceback:

```
(.venv) mdickinson@mirzakhani bokehbug % bokeh serve custom_extension.py --show 
2024-03-05 08:02:43,112 Starting Bokeh server version 3.3.4 (running on Tornado 6.4)
2024-03-05 08:02:43,117 User authentication hooks NOT provided (default user enabled)
2024-03-05 08:02:43,122 Bokeh app running at: http://localhost:5006/custom_extension
2024-03-05 08:02:43,122 Starting Bokeh server with process id: 11374
Compilation failed:

custom.ts:4:1 - error TS6133: 'p' is declared but its value is never read.

4 import * as p from "core/properties"
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
custom.ts:13:29 - error TS2339: Property 'slider' does not exist on type 'UIElement'.

13     this.connect(this.model.slider.change, () => {
                               ~~~~~~
custom.ts:32:22 - error TS2339: Property 'text' does not exist on type 'UIElement'.

32     }, `${this.model.text}: ${this.model.slider.value}`))
                        ~~~~
custom.ts:32:42 - error TS2339: Property 'slider' does not exist on type 'UIElement'.

32     }, `${this.model.text}: ${this.model.slider.value}`))
                                            ~~~~~~
custom.ts:56:17 - error TS2702: 'Custom' only refers to a type, but is being used as a namespace here.

56     this.define<Custom.Props>(({String, Ref}) => ({
                   ~~~~~~
custom.ts:56:33 - error TS7031: Binding element 'String' implicitly has an 'any' type.

56     this.define<Custom.Props>(({String, Ref}) => ({
                                   ~~~~~~
custom.ts:56:41 - error TS7031: Binding element 'Ref' implicitly has an 'any' type.

56     this.define<Custom.Props>(({String, Ref}) => ({
                                           ~~~
custom.ts:58:21 - error TS2304: Cannot find name 'Slider'.

58       slider: [ Ref(Slider) ],
                       ~~~~~~
```


### Example code

```Python
# The `custom.py` file and `custom.ts` file are identical to those on the documentation page.
# Here's the modified `custom_extension.py` that I used.

from bokeh.layouts import column
from bokeh.models import Slider

from custom import Custom

slider = Slider(start=0, end=10, step=0.1, value=0, title="value")

custom = Custom(text="Special Slider Display", slider=slider)

layout = column(slider, custom)
```


### Stack traceback or browser console output

The issue seems very similar to that raised in #9587 (long since closed). The documentation page referred to in that issue no longer exists, but the code in the new documentation appears to be derived from that code.

Apart from the TS compilation errors above, I think there's at least one other error that prevents the compiled extension from working correctly: the line `static __name__ = "Surface3d"` in the `Custom` class in the TypeScript code should read `static __name__ = "Custom"`



### Screenshots

_No response_

--- a/bokehjs/test/devtools/server.ts
+++ b/bokehjs/test/devtools/server.ts
@@ -114,7 +114,10 @@ app.get("/examples/:name", async (req, res) => {
   res.status(404).send("No such example")
 })
 
-async function build_example(path: string): Promise<string | null> {
+type Resources = "server" | "cdn"
+type BuildOptions = {dev?: boolean, resources?: Resources}
+
+async function build_example(path: string, options: BuildOptions = {}): Promise<string | null> {
   const code = `\
 __file__ = "${path}"
 
@@ -133,7 +136,8 @@ with open(__file__, "rb") as example:
 
   const env = {
     ...process.env,
-    BOKEH_DEV: "true",
+    BOKEH_DEV: (options.dev ?? true) ? "true" : "false",
+    BOKEH_RESOURCES: options.resources ?? "server",
     BOKEH_DEFAULT_SERVER_HOST: host,
     BOKEH_DEFAULT_SERVER_PORT: `${port}`,
   }
@@ -219,7 +223,7 @@ app.get("/bokeh/examples/:path(*)", async (req, res) => {
     return
   }
 
-  const error = await build_example(py_path)
+  const error = await build_example(py_path, {dev: argv.dev, resources: argv.resources as Resources})
   if (error != null) {
     res.status(200).render("test/devtools/bokeh_example.html", {title: py_path, contents: error})
     return
@@ -242,6 +246,8 @@ process.once("SIGTERM", () => {
 const argv = yargs(process.argv.slice(2)).options({
   host: {type: "string", default: "127.0.0.1"},
   port: {type: "number", default: 5777},
+  dev: {type: "boolean", default: true},
+  resources: {type: "string", choices: ["server", "cdn"] as const, default: "server"},
 }).parseSync()
 
 const {host, port} = argv

--- a/docs/bokeh/source/docs/user_guide/advanced/extensions.rst
+++ b/docs/bokeh/source/docs/user_guide/advanced/extensions.rst
@@ -79,69 +79,8 @@ Here's an annotated TypeScript implementation for ``Custom`` and its
 ``CustomView``. For built-in models, this type of code is included directly in
 the final BokehJS scripts.
 
-.. code-block:: typescript
-
-    import {UIElement, UIElementView} from "models/ui/ui_element"
-
-    import {div} from "core/dom"
-    import * as p from "core/properties"
-
-    export class CustomView extends UIElementView {
-
-      override connect_signals(): void {
-        super.connect_signals()
-
-        // Set BokehJS listener so that the program can process new data when
-        // the slider has a change event.
-        this.connect(this.model.slider.change, () => {
-          this.render()
-        })
-      }
-
-      override render(): void {
-        // BokehJS views create <div> elements by default. These are accessible
-        // as ``this.el``. Many Bokeh views ignore the default <div> and
-        // instead do things like draw to the HTML canvas. In this case though,
-        // the program changes the contents of the <div> based on the current
-        // slider value.
-        super.render()
-
-        this.shadow_el.appendChild(div({
-          style: {
-            padding: '2px',
-            color: '#b88d8e',
-            backgroundColor: '#2a3153',
-          },
-        }, `${this.model.text}: ${this.model.slider.value}`))
-      }
-    }
-
-    export class Custom extends UIElement {
-      slider: {value: string}
-
-      // Generally, the ``__name__`` class attribute should match the name of
-      // the corresponding Python class exactly. TypeScript matches the name
-      // automatically during compilation, so, barring some special cases, you
-      // don't have to do this manually. This helps avoid typos, which stop
-      // serialization/deserialization of the model.
-      static __name__ = "Surface3d"
-
-      static {
-        // If there is an associated view, this is typically boilerplate.
-        this.prototype.default_view = CustomView
-
-        // The this.define() block adds corresponding "properties" to the JS
-        // model. These should normally line up 1-1 with the Python model
-        // class. Most property types have counterparts. For example,
-        // bokeh.core.properties.String will correspond to ``String`` in the
-        // JS implementation. Where JS lacks a given type, you can use
-        // ``p.Any`` as a "wildcard" property type.
-        this.define<Custom.Props>(({String, Ref}) => ({
-          text:   [ String ],
-          slider: [ Ref(Slider) ],
-        }))
-      }
-    }
+.. literalinclude:: /../../../examples/advanced/extensions/custom.ts
+   :language: typescript
 
 .. _ug_advanced_extensions_structure_putting_together:
 

--- a/examples/advanced/extensions/custom.ts
+++ b/examples/advanced/extensions/custom.ts
@@ -1,0 +1,75 @@
+import {UIElement, UIElementView} from "models/ui/ui_element"
+import {Slider} from "models/widgets/sliders/slider"
+import {div} from "core/dom"
+import * as p from "core/properties"
+
+export class CustomView extends UIElementView {
+  declare model: Custom
+
+  private content_el: HTMLElement
+
+  override connect_signals(): void {
+    super.connect_signals()
+    this.connect(this.model.slider.change, () => this._update_text())
+  }
+
+  override render(): void {
+    // BokehJS views create <div> elements by default. These are accessible
+    // as ``this.el``. Many Bokeh views ignore the default <div> and
+    // instead do things like draw to the HTML canvas. In this case though,
+    // the program changes the contents of the <div> based on the current
+    // slider value.
+    super.render()
+
+    this.content_el = div({style: {
+      textAlign: "center",
+      fontSize: "1.2em",
+      padding: "2px",
+      color: "#b88d8e",
+      backgroundColor: "#2a3153",
+    }})
+    this.shadow_el.append(this.content_el)
+
+    this._update_text()
+  }
+
+  private _update_text(): void {
+    this.content_el.textContent = `${this.model.text}: ${this.model.slider.value}`
+  }
+}
+
+export namespace Custom {
+  export type Attrs = p.AttrsOf<Props>
+
+  export type Props = UIElement.Props & {
+    text: p.Property<string>
+    slider: p.Property<Slider>
+  }
+}
+
+export interface Custom extends Custom.Attrs {}
+
+export class Custom extends UIElement {
+  declare properties: Custom.Props
+  declare __view_type__: CustomView
+
+  constructor(attrs?: Partial<Custom.Attrs>) {
+    super(attrs)
+  }
+
+  static {
+    // If there is an associated view, this is typically boilerplate.
+    this.prototype.default_view = CustomView
+
+    // The this.define() block adds corresponding "properties" to the JS
+    // model. These should normally line up 1-1 with the Python model
+    // class. Most property types have counterparts. For example,
+    // bokeh.core.properties.String will correspond to ``String`` in the
+    // JS implementation. Where JS lacks a given type, you can use
+    // ``p.Any`` as a "wildcard" property type.
+    this.define<Custom.Props>(({Str, Ref}) => ({
+      text:   [ Str, "Custom text" ],
+      slider: [ Ref(Slider) ],
+    }))
+  }
+}

--- a/examples/advanced/extensions/putting_together.py
+++ b/examples/advanced/extensions/putting_together.py
@@ -2,89 +2,11 @@ from bokeh.core.properties import Instance, Required, String
 from bokeh.io import show
 from bokeh.layouts import column
 from bokeh.models import Slider, UIElement
-from bokeh.util.compiler import TypeScript
 
-CODE = """
-import {UIElement, UIElementView} from "models/ui/ui_element"
-import {Slider} from "models/widgets/sliders/slider"
-import {div} from "core/dom"
-import * as p from "core/properties"
-
-export class CustomView extends UIElementView {
-  declare model: Custom
-
-  private content_el: HTMLElement
-
-  override connect_signals(): void {
-    super.connect_signals()
-    this.connect(this.model.slider.change, () => this._update_text())
-  }
-
-  override render(): void {
-    // BokehJS views create <div> elements by default. These are accessible
-    // as ``this.el``. Many Bokeh views ignore the default <div> and
-    // instead do things like draw to the HTML canvas. In this case though,
-    // the program changes the contents of the <div> based on the current
-    // slider value.
-    super.render()
-
-    this.content_el = div({style: {
-      textAlign: "center",
-      fontSize: "1.2em",
-      padding: "2px",
-      color: "#b88d8e",
-      backgroundColor: "#2a3153",
-    }})
-    this.shadow_el.appendChild(this.content_el)
-
-    this._update_text()
-  }
-
-  private _update_text(): void {
-    this.content_el.textContent = `${this.model.text}: ${this.model.slider.value}`
-  }
-}
-
-export namespace Custom {
-  export type Attrs = p.AttrsOf<Props>
-
-  export type Props = UIElement.Props & {
-    text: p.Property<string>
-    slider: p.Property<Slider>
-  }
-}
-
-export interface Custom extends Custom.Attrs {}
-
-export class Custom extends UIElement {
-  declare properties: Custom.Props
-  declare __view_type__: CustomView
-
-  constructor(attrs?: Partial<Custom.Attrs>) {
-    super(attrs)
-  }
-
-  static {
-    // If there is an associated view, this is typically boilerplate.
-    this.prototype.default_view = CustomView
-
-    // The this.define() block adds corresponding "properties" to the JS
-    // model. These should normally line up 1-1 with the Python model
-    // class. Most property types have counterparts. For example,
-    // bokeh.core.properties.String will correspond to ``String`` in the
-    // JS implementation. Where JS lacks a given type, you can use
-    // ``p.Any`` as a "wildcard" property type.
-    this.define<Custom.Props>(({Str, Ref}) => ({
-      text:   [ Str, "Custom text" ],
-      slider: [ Ref(Slider) ],
-    }))
-  }
-}
-"""
 
 class Custom(UIElement):
 
-    __implementation__ = TypeScript(CODE)
+    __implementation__ = "custom.ts"
 
     text = String(default="Custom text")
 

